Makefile.core.mk

```diff
@@ -108,6 +108,71 @@ $(shell mkdir -p $(dir $(JUNIT_OUT)))
 $(TARGET_OUT):
 	@mkdir -p $@
 
+# scratch dir: this shouldn't be simply 'docker' since that's used for docker.save to store tar.gz files
+ISTIO_DOCKER:=${TARGET_OUT_LINUX}/docker_temp
+
+# scratch dir for building isolated images. Please don't remove it again - using
+# ISTIO_DOCKER results in slowdown, all files (including multiple copies of envoy) will be
+# copied to the docker temp container - even if you add only a tiny file, >1G of data will
+# be copied, for each docker image.
+DOCKER_BUILD_TOP:=${TARGET_OUT_LINUX}/docker_build
+DOCKERX_BUILD_TOP:=${TARGET_OUT_LINUX}/dockerx_build
+
+# dir where tar.gz files from docker.save are stored
+ISTIO_DOCKER_TAR:=${TARGET_OUT_LINUX}/release/docker
+
+# Populate the git version for istio/proxy (i.e. Envoy)
+ifeq ($(PROXY_REPO_SHA),)
+  export PROXY_REPO_SHA:=$(shell grep PROXY_REPO_SHA istio.deps  -A 4 | grep lastStableSHA | cut -f 4 -d '"')
+endif
+
+# Envoy binary variables Keep the default URLs up-to-date with the latest push from istio/proxy.
+
+export ISTIO_ENVOY_BASE_URL ?= https://storage.googleapis.com/istio-build/proxy
+
+# Use envoy as the sidecar by default
+export SIDECAR ?= envoy
+
+# OS-neutral vars. These currently only work for linux.
+export ISTIO_ENVOY_VERSION ?= ${PROXY_REPO_SHA}
+export ISTIO_ENVOY_DEBUG_URL ?= $(ISTIO_ENVOY_BASE_URL)/envoy-debug-$(ISTIO_ENVOY_VERSION).tar.gz
+export ISTIO_ENVOY_CENTOS_DEBUG_URL ?= $(ISTIO_ENVOY_BASE_URL)/envoy-centos-debug-$(ISTIO_ENVOY_VERSION).tar.gz
+export ISTIO_ENVOY_RELEASE_URL ?= $(ISTIO_ENVOY_BASE_URL)/envoy-alpha-$(ISTIO_ENVOY_VERSION).tar.gz
+export ISTIO_ENVOY_CENTOS_RELEASE_URL ?= $(ISTIO_ENVOY_BASE_URL)/envoy-centos-alpha-$(ISTIO_ENVOY_VERSION).tar.gz
```

Comment:
Sorry if i miss something, are these makefile changes related to the SGX work? Do we really need them?

Reply:
Ahh, this was automatically migrated during cherry pick. Istio has modified the way docker builds before, so I think we don't need these anymore, thanks for correction.